### PR TITLE
tetragon: improve how we read process info during startup

### PR DESCRIPTION
**Pull request by tixxdz**
Signed-off-by: Djalal Harouni <tixxdz@gmail.com>

### CODE REVIEW

File: pkg/reader/proc/proc.go [hunk 1]
```diff
@@ -78,6 +150,38 @@
...
+	if err != nil {
+		return InvalidUid, InvalidUid, err
+	}
+
+	euid, err := strconv.Atoi(status.Uids[1])
+	if err != nil {
+		return InvalidUid, InvalidUid, err
+	}
+
+	return uint32(ruid), uint32(euid), nil
```
[the remark this answers is not in the document]
Comment (tixxdz) [reply pkg/reader/proc/proc.go:170]: we explicitly wanted the conversion... changed to suggested, let's hope it does not trigger

[the remark this answers is not in the document]
Comment (tixxdz) [reply pkg/reader/proc/proc.go:170]: We use the 4294967295 (-1 as an unsigned integer) as invalid uids, but lets just fix to what suggested

**Comment by kkourt** [comment]
Thanks Djalal. The static tests are failing. Can you have a look? If it makes sense to fix them, let's do it. Otherwise, I'd suggest adding a comment in the PR about why the test is failing and why we do not want to fix it.

**Comment by kkourt** [comment]
Thanks!

The static tests are failing. It's just a minior suggestion from revive, but this means that all the tests after that are not running. Can you please do what the linter suggests? You can see the errors in the "Files Changed" tab, or by running `make check`.

**Comment by tixxdz** [comment]
> Thanks!
> 
> The static tests are failing. It's just a minor suggestion from revive, but this means that all the tests after that are not running. Can you please do what the linter suggests? You can see the errors in the "Files Changed" tab, or by running `make check`.

Thank you @kkourt totally forgot about the make check... updated!

File: bpf/process/bpf_process_event.h [hunk 2]
```diff
@@ -457,7 +457,8 @@ get_caps(struct msg_capabilities *msg, struct task_struct *task)
 {
 	const struct cred *cred;
 
-	probe_read(&cred, sizeof(cred), _(&task->real_cred));
```
Comment (jrfastab) [inline bpf/process/bpf_process_event.h:460]: can you add some notes to the commit message on what cred and real_cred are and why one is preferred to the other.

**Comment by jrfastab** [comment]
PR LGTM please fix up commit message on the BPF patch so folks can reason about why cred vs real_cred and then fixup the static check or tell us it doesn't make sense to fix up for some reason.

**Comment by tixxdz** [comment]
> PR LGTM please fix up commit message on the BPF patch so folks can reason about why cred vs real_cred and then fixup the static check or tell us it doesn't make sense to fix up for some reason.

Ok done John, thanks for the review! I fixed the static check, I don't get the checkpatch complain...

**Comment by tixxdz** [comment]
@jrfastab please merge this one, the 4.19 test that failed is fixed by @kkourt here: https://github.com/cilium/tetragon/pull/531

**Pull request merged**